auth_mellon_config.c

```diff
@@ -1886,6 +1998,45 @@ void *auth_mellon_server_config(apr_pool_t *p, server_rec *s)
     apr_pool_userdata_set(mod, key, apr_pool_cleanup_null, p);
 
     srv->mc = mod;
+
     return srv;
 }
 
+/* This function merges two am_srv_cfg_rec structures.
+ * It will try to inherit from the base where possible.
+ *
+ * Parameters:
+ *  apr_pool_t *p        The pool we should allocate memory from.
+ *  void *base           The original structure.
+ *  void *add            The structure we should add to base.
+ *
+ * Returns:
+ *  The merged structure.
+ */
+void *auth_mellon_srv_merge(apr_pool_t *p, void *base, void *add)
+{
+    am_srv_cfg_rec *base_cfg = (am_srv_cfg_rec *)base;
+    am_srv_cfg_rec *new_cfg;
+
+    new_cfg = (am_srv_cfg_rec *)apr_palloc(p, sizeof(*new_cfg));
+
+    new_cfg->mc = base_cfg->mc;
+
+#ifdef WITH_DIAGNOSTICS
+    am_srv_cfg_rec *add_cfg = (am_srv_cfg_rec *)add;
+    new_cfg->diag_cfg.filename = (add_cfg->diag_cfg.filename !=
+                                  default_diag_filename ?
+                                  add_cfg->diag_cfg.filename :
+                                  base_cfg->diag_cfg.filename);
+
+    new_cfg->diag_cfg.flags = (add_cfg->diag_cfg.flags !=
+                               default_diag_flags ?
+                               add_cfg->diag_cfg.flags :
+                               base_cfg->diag_cfg.flags);
+
+    new_cfg->diag_cfg.dir_cfg_emitted = apr_table_make(p, 0);
+
```

Comment:
When testing this code, I got a segmentation fault that i traced to `new_cfg->diag_cfg.fd` not being initialized in this function.

(I am guessing that I got that error because I dropped the diagnostics options inside a `<VirtualHost>`-entry.)

Reply:
Thanks for finding this, it was definitely a problem. I had mistakenly omitted initializing the fd in the merge function. Fixed.